File: frontend/lib/models/event.dart
import 'package:flutter/material.dart';

/// Event data model matching backend API
class Event {
  final String id;
  final String title;
  final String summary;
  final int heatScore;
  final String sentiment;
  final List<String> emotionTags;
  final String region;
  final List<EventSource> sources;
  final List<String> keywords;
  final DateTime createdAt;
  final DateTime updatedAt;

  Event({
    required this.id,
    required this.title,
    required this.summary,
    required this.heatScore,
    required this.sentiment,
    required this.emotionTags,
    required this.region,
    required this.sources,
    required this.keywords,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Event.fromJson(Map<String, dynamic> json) {
    return Event(
      id: json['id'] ?? '',
      title: json['title'] ?? '',
      summary: json['summary'] ?? '',
      heatScore: json['heat_score'] ?? 0,
      sentiment: json['sentiment'] ?? 'neutral',
      emotionTags: List<String>.from(json['emotion_tags'] ?? []),
      region: json['region'] ?? 'china',
      sources: (json['sources'] as List?)
          ?.map((s) => EventSource.fromJson(s))
          .toList() ?? [],
      keywords: List<String>.from(json['keywords'] ?? []),
      createdAt: DateTime.tryParse(json['created_at'] ?? '') ?? DateTime.now(),
      updatedAt: DateTime.tryParse(json['updated_at'] ?? '') ?? DateTime.now(),
    );
  }

  /// Get sentiment color
  Color get sentimentColor {
    switch (sentiment) {
      case 'positive':
        return const Color(0xFF4CAF50);
      case 'negative':
        return const Color(0xFFF44336);
      case 'mixed':
        return const Color(0xFFFF9800);
      default:
        return const Color(0xFF9E9E9E);
    }
  }

  /// Get sentiment label
  String get sentimentLabel {
    switch (sentiment) {
      case 'positive':
        return '正面';
      case 'negative':
        return '负面';
      case 'mixed':
        return '争议';
      default:
        return '中性';
    }
  }
}

class EventSource {
  final String platform;
  final String? url;
  final String? author;

  EventSource({
    required this.platform,
    this.url,
    this.author,
  });

  factory EventSource.fromJson(Map<String, dynamic> json) {
    return EventSource(
      platform: json['platform'] ?? '',
      url: json['url'],
      author: json['author'],
    );
  }
}

class EventList {
  final int total;
  final String region;
  final List<Event> events;

  EventList({
    required this.total,
    required this.region,
    required this.events,
  });

  factory EventList.fromJson(Map<String, dynamic> json) {
    return EventList(
      total: json['total'] ?? 0,
      region: json['region'] ?? 'china',
      events: (json['events'] as List?)
          ?.map((e) => Event.fromJson(e))
          .toList() ?? [],
    );
  }
}

File: backend/app/models/event.py
"""Event Data Models"""

from datetime import datetime
from enum import Enum
from typing import Optional, List
from pydantic import BaseModel, Field


class SentimentType(str, Enum):
    """Sentiment type enumeration."""
    POSITIVE = "positive"
    NEGATIVE = "negative"
    NEUTRAL = "neutral"
    MIXED = "mixed"


class EventSource(BaseModel):
    """Event source information."""
    platform: str = Field(..., description="Source platform (weibo, xiaohongshu, etc.)")
    url: Optional[str] = Field(None, description="Original URL")
    author: Optional[str] = Field(None, description="Original author/account")


class Event(BaseModel):
    """Hot event data model."""
    id: str = Field(..., description="Unique event ID")
    title: str = Field(..., description="Event title")
    summary: str = Field(..., description="AI-generated summary")
    heat_score: int = Field(..., ge=0, le=100, description="Heat score (0-100)")
    sentiment: SentimentType = Field(default=SentimentType.NEUTRAL, description="Sentiment analysis result")
    emotion_tags: List[str] = Field(default_factory=list, description="Emotion tags (anger, joy, fear, etc.)")
    region: str = Field(default="china", description="Region code")
    sources: List[EventSource] = Field(default_factory=list, description="Source list")
    keywords: List[str] = Field(default_factory=list, description="Keywords")
    created_at: datetime = Field(default_factory=datetime.now, description="Created time")
    updated_at: datetime = Field(default_factory=datetime.now, description="Updated time")

    class Config:
        json_schema_extra = {
            "example": {
                "id": "evt_20240104_001",
                "title": "某科技公司发布新品引发热议",
                "summary": "该公司发布的新产品在社交媒体上引发广泛讨论，用户反馈褒贬不一...",
                "heat_score": 85,
                "sentiment": "mixed",
                "emotion_tags": ["excitement", "curiosity", "skepticism"],
                "region": "china",
                "sources": [
                    {"platform": "weibo", "url": "https://weibo.com/...", "author": "科技博主"}
                ],
                "keywords": ["科技", "新品", "发布会"],
                "created_at": "2024-01-04T10:00:00",
                "updated_at": "2024-01-04T10:00:00"
            }
        }


class EventList(BaseModel):
    """Event list response model."""
    total: int = Field(..., description="Total event count")
    region: str = Field(..., description="Region code")
    events: List[Event] = Field(default_factory=list, description="Event list")
